src/packages/frontend/projects/use-bookmarked-projects.ts

```diff
@@ -0,0 +1,154 @@
+/*
+ *  This file is part of CoCalc: Copyright © 2024 Sagemath, Inc.
+ *  License: MS-RSL – see LICENSE.md for details
+ */
+
+/**
+// To debug bookmarked projects in the browser console:
+c = cc.client.conat_client
+bm = await c.dkv({account_id: cc.client.account_id, name: 'bookmarks'})
+// Check all bookmark data
+console.log('All bookmarks:', bm.getAll())
+// Check bookmarked projects
+console.log('Bookmarked projects:', bm.get("projects"))
+// Set bookmarked projects
+bm.set("projects", ['project_id_1', 'project_id_2'])
+// Listen to changes
+bm.on('change', (e) => console.log('Bookmark change:', e))
+ */
+
+import { sortBy, uniq } from "lodash";
+import { useEffect, useRef, useState } from "react";
+
+import { redux } from "@cocalc/frontend/app-framework";
+import { webapp_client } from "@cocalc/frontend/webapp-client";
+
+const CONAT_BOOKMARKS_KEY = "bookmarks";
+const PROJECTS_KEY = "projects";
+
+export type BookmarkedProjects = string[]; // array of project UUIDs
+
+// Bookmarked projects are now managed entirely through conat with in-memory state.
+// No local storage dependency - conat handles synchronization and persistence.
+export function useBookmarkedProjects() {
+  const [bookmarkedProjects, setBookmarkedProjects] =
+    useState<BookmarkedProjects>([]);
+  const [bookmarks, setBookmarks] = useState<any>(null);
+  const [isInitialized, setIsInitialized] = useState(false);
+
+  // Use ref to store stable listener function
+  const listenerRef = useRef<
+    | ((changeEvent: {
+        key: string;
+        value?: Record<string, any>;
+        prev?: Record<string, any>;
+      }) => void)
+    | null
+  >(null);
+
+  // Initialize conat bookmarks and set up listeners
+  useEffect(() => {
```

Comment:
Maybe useAsyncEffect would be slightly better for this?   Search our code for it...